src/Agent.Listener/SelfUpdater.cs

```diff
@@ -236,8 +245,38 @@ private async Task DownloadLatestAgent(CancellationToken token)
                             }
 
                             Trace.Info($"Download agent: finished download");
-                            downloadSucceeded = true;
-                            break;
+
+                            // Determine the service deployment type based on connection data. (Hosted/OnPremises)
+                            bool isHostedServer = await Validators.IsHostedServer(_serverUrl, _creds, _locationServer);
+
+                            if (!isHostedServer)
+                            {
+                                Trace.Info($"Skipping checksum validation for On-Premises solution");
+                                downloadSucceeded = true;
+                            }
+                            else if (string.IsNullOrEmpty(_targetPackage.HashValue))
+                            {
+                                Trace.Warning($"Unable to perform the necessary checksum validation since the target package hash is missed");
+                            }
+                            else
+                            {
+                                string archiveHash = IOUtil.GetFileHash(archiveFile);
```

Comment:
Please move validation logic to separate method

Reply:
Do we have a fallback and agent knob for validation?

Reply:
Created new agent knob `AZP_AGENT_HASH_VALIDATION_DISABLED` — `false` by default